// @betting-api/fonbet/src/football/api.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const invariant_1 = __importDefault(require("invariant"));
const client_1 = __importDefault(require("../../client"));
async function getLive(id) {
    invariant_1.default(id, `fonbet.getLive => invalid id: ${id}`);
    const { data } = await client_1.default.get(`/football/live/${id}`);
    return data;
}
async function getLiveAll(params) {
    const { data } = await client_1.default.get('/football/live/all', {
        params,
    });
    return data;
}
async function getMatch(id) {
    invariant_1.default(id, `fonbet.getMatch => invalid id: ${id}`);
    /** live or prematch * */
    const { data } = await client_1.default.get(`/football/match/${id}`);
    return data;
}
async function getPreMatch(id) {
    invariant_1.default(id, `fonbet.getPreMatch => invalid id: ${id}`);
    const { data } = await client_1.default.get(`/football/line/${id}`);
    return data;
}
async function getPreMatchAll(params) {
    const { data } = await client_1.default.get('/football/line/all', {
        params,
    });
    return data;
}
async function getPreMatchLeagues() {
    const { data } = await client_1.default.get('/football/line/leagues');
    return data;
}
async function getPreMatchLeagueMatches(league_id) {
    invariant_1.default(league_id, `fonbet.getPreMatchLeagueMatches => invalid id: ${league_id}`);
    const { data } = await client_1.default.get(`/football/line/league/${league_id}/matches`);
    return data;
}
async function getLiveLeagues() {
    const { data } = await client_1.default.get('/football/live/leagues');
    return data;
}
async function getLiveLeagueMatches(league_id) {
    invariant_1.default(league_id, `fonbet.getLiveLeagueMatches => invalid id: ${league_id}`);
    const { data } = await client_1.default.get(`/football/live/league/${league_id}/matches`);
    return data;
}
async function getInfo() {
    const { data } = await client_1.default.get('/football/info');
    return data;
}
exports.default = {
    getLive,
    getLiveAll,
    getLiveLeagues,
    getLiveLeagueMatches,
    getPreMatch,
    getPreMatchAll,
    getPreMatchLeagues,
    getPreMatchLeagueMatches,
    getMatch,
    getInfo,
};
